src/Microsoft.AspNet.Hosting/Internal/HostingEngine.cs

```diff
@@ -87,6 +88,7 @@ public virtual IDisposable Start()
                 _applicationLifetime.NotifyStopping();
                 server.Dispose();
                 _applicationLifetime.NotifyStopped();
+                loggerFactory.Dispose();
```

Comment:
It feels odd to just dispose this service here. Not sure if there are any other services which need similar requirement.